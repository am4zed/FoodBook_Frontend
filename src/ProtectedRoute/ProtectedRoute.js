import React, { Component } from "react"
import { Route } from "react-router-dom"

const ProtectedRoute = (props) => {
    const { component: Component, path, validatingSession } = props
    return (
        <Route path={path} render={() => {
            if (validatingSession) return <h3 className="text-center">Style me. This is validating session</h3>;
            return <Component />
        }} />
    );
}

export default ProtectedRoute;