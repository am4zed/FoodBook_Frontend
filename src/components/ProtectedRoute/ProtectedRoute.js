import React from "react"
import { Route } from "react-router-dom"

const ProtectedRoute = (props) => {
    const { component: Component, path, validatingSession, render } = props
    return (
        <Route path={path} render={() => {
            if (validatingSession) return <h3 className="text-center">Style me. This is validating session</h3>;
            return Component ? <Component /> : render();
        }} />
    );
}

export default ProtectedRoute;