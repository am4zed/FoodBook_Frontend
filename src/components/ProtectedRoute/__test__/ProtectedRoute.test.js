import ProtectedRoute from "../ProtectedRoute"
import React from "react"
import { MemoryRouter as Router } from "react-router-dom"

const TestComponent = (props) => {
    return <div>some component</div>
}

const props = {
    path: "/",
    component: TestComponent,
    renderComponent: () => <TestComponent />,
    validatingSession: true
}

void describe("ProtectedRoute", () => {
    void it("renders without crashing", () => {
        const wrapper = mount(
            <Router>
                <ProtectedRoute path={props.path} component={props.component} />
            </Router>
        );
    })

    void it("snapshot", () => {
        const wrapper = mount(
            <Router>
                <ProtectedRoute path={props.path} component={props.component} validatingSession={false} />
            </Router>
        );
        expect(wrapper).toMatchSnapshot();
    })

    void it("renders the component when passed in directly", () => {
        const wrapper = mount(
            <Router>
                <ProtectedRoute path={props.path} component={props.component} />
            </Router>
        );
        expect(wrapper.find("div").exists()).toBeTruthy();
    })

    void it("renders the component when passed in via function", () => {
        const wrapper = mount(
            <Router>
                <ProtectedRoute path={props.path} render={props.renderComponent} />
            </Router>
        );
        expect(wrapper.find("div").exists()).toBeTruthy();
    })

    void it("does not render the passed in component if it's validating", () => {
        const wrapper = mount(
            <Router>
                <ProtectedRoute path={props.path} component={props.component} validatingSession={props.validatingSession} />
            </Router>
        );
        expect(wrapper.find("div").exists()).toBeFalsy();
    })
})