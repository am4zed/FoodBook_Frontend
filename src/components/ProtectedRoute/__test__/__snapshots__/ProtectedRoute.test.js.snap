// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute snapshot 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "rb4yhe",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "rt278c",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ProtectedRoute
      component={[Function]}
      path="/"
      validatingSession={false}
    >
      <Route
        path="/"
        render={[Function]}
      >
        <TestComponent>
          <div>
            some component
          </div>
        </TestComponent>
      </Route>
    </ProtectedRoute>
  </Router>
</MemoryRouter>
`;
